app/models/hmpps_api/offender.rb

```diff
@@ -1,20 +1,289 @@
+      end
+    end
+
+    def pom_supporting?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).custody_supporting?
+      else
+        @responsibility_override.value == Responsibility::PROBATION
+      end
+    end
+
+    def com_responsible?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).community_responsible?
+      else
+        @responsibility_override.value == Responsibility::PROBATION
+      end
+    end
+
+    def com_supporting?
+      if @responsibility_override.nil?
+        HandoverDateService.handover(self).community_supporting?
+      else
+        # Overrides to prison aren't actually possible in the UI
+        # If they were, we'd somehow need to decide whether COM is supporting or not involved
+        @responsibility_override.value == Responsibility::PRISON
+      end
+    end
+
+    def sentence_start_date
+      sentence.sentence_start_date
+    end
+
+    def full_name
+      "#{last_name}, #{first_name}".titleize
+    end
+
+    def full_name_ordered
+      "#{first_name} #{last_name}".titleize
+    end
+
+    def responsibility_override?
+      @responsibility_override.present?
+    end
+
+    # This list must only contain fields that are both supplied by
+    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//prisoners/getPrisonersOffenderNo
+    # and also by
+    # https://api-dev.prison.service.justice.gov.uk/swagger-ui.html#//locations/getOffendersAtLocationDescription
+    def initialize(api_payload, search_payload, category:, latest_temp_movement:, complexity_level:, booking_id:, prison_id:)
+      # It is expected that this method will be called by the subclass which
+      # will have been given a payload at the class level, and will call this
+      # method from it's own internal from_json
+      @first_name = api_payload.fetch('firstName')
+      @last_name = api_payload.fetch('lastName')
+      @offender_no = api_payload.fetch('offenderNo')
+      @convicted_status = api_payload['convictedStatus']
+      @date_of_birth = Date.parse(api_payload.fetch('dateOfBirth'))
+      @latest_temp_movement = latest_temp_movement
+      @cell_location = search_payload['cellLocation']
+      @complexity_level = complexity_level
+      @category = category
+
+      @booking_id = booking_id
+      @prison_id = prison_id
+    end
+
+    def handover_start_date
+      handover.start_date
+    end
+
+    def handover_reason
+      handover.reason_text
+    end
+
+    def responsibility_handover_date
+      handover.handover_date
+    end
+
+    def load_case_information(record)
+      return if record.blank?
+
+      # Hold on to the CaseInformation record so we can reference it later
+      @case_information = record
+
+      # This is separate from @case_information so the rest of this class doesn't need to know about the Active Record association
+      @responsibility_override = record.responsibility
+    end
```

Comment:
It might also be nice to consider this another 'constructor' method and keep it at the top of the file alongside the `initialize` method. It's another method that puts something 'in' to the object for the other methods to use.